edelphi/src/main/java/fi/metatavu/edelphi/pages/panel/admin/report/util/ChartModelProvider.java

```diff
@@ -123,15 +122,16 @@ public static Chart createBarChart(String chartCaption, String xLabel, List<Stri
     ySeries.getSeriesPalette().update(ColorDefinitionImpl.create(0, 153, 255));
     ySeries.getSeries().add(bs1);
 
-    addMarkerLine("A", average, 0.5d, xAxisPrimary);
-    addMarkerLine("Q1", q1, 0.5d, xAxisPrimary);
-    addMarkerLine("Q3", q3, 0.5d, xAxisPrimary);
+    int size = categoryCaptions.size() - 1;
+    addMarkerLine("A", translateMarkerToRange(size, min, max, average), 0.5d, xAxisPrimary);
+    addMarkerLine("Q1", translateMarkerToRange(size, min, max, q1), 0.5d, xAxisPrimary);
+    addMarkerLine("Q3", translateMarkerToRange(size, min, max, q3), 0.5d, xAxisPrimary);
 
     return cwaBar;
   }
 
-  public static Chart createBarChartHorizontal(String chartCaption, String xLabel, List<String> categoryCaptions, List<Double> values, Double average, Double q1, Double q3) {
-    ChartWithAxes chart = (ChartWithAxes) createBarChart(chartCaption, xLabel, categoryCaptions, values, average, q1, q3);
+  public static Chart createBarChartHorizontal(String chartCaption, String xLabel, List<String> categoryCaptions, List<Double> values, Double min, Double max, Double average, Double q1, Double q3) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)